### DISCUSSION TITLE
Is use of string templates discouraged due to difficulty in localising them?

### DISCUSSION THREAD

**Discussion by aral** [elementary/docs#146] [Q&A]
Reference: https://github.com/elementary/docs/blob/master/writing-apps/code-style.md#string-formatting

It would be good to provide a reason for why string templates are discouraged, given their ergonomic advantages over remembering every `%` code on the planet.

I haven’t been able to find any information on how translations work/don’t work with string templates but I’m assuming they are complicated at the very least. Is this the reason why they’re discouraged?

**Comment by lenemter** [comment]
Closing as duplicate of https://github.com/elementary/docs/issues/147
